src/java.base/share/classes/java/time/ZoneOffset.java

```diff
@@ -428,8 +428,10 @@ public static ZoneOffset ofTotalSeconds(int totalSeconds) {
             ZoneOffset result = SECONDS_CACHE.get(totalSecs);
             if (result == null) {
                 result = new ZoneOffset(totalSeconds);
-                SECONDS_CACHE.putIfAbsent(totalSecs, result);
-                result = SECONDS_CACHE.get(totalSecs);
+                ZoneOffset prev = SECONDS_CACHE.putIfAbsent(totalSecs, result);
+                if (prev != null) {
+                    result = prev;
+                }
                 ID_CACHE.putIfAbsent(result.getId(), result);
```

Comment:
Feel free to ignore this one, but you could still have this be a `computeIfAbsent` and put the ID_CACHE.putIfAbsent part in the lambda. Yeah, there'll be more work done inside of `computeIfAbsent` but I think it's an acceptable tradeoff for a more comprehensible code.

Reply:
udpated